bridge-token-factory/src/lib.rs

```diff
@@ -342,6 +434,212 @@ impl BridgeTokenFactory {
+    }
+
+    pub fn get_withdraw_token_fee_percentage(
+        &self,
+        token: &String,
+    ) -> Option<WithdrawFeePercentage> {
+        self.withdraw_fee_percentage.get(token)
+    }
+
+    // Returns fee percentage for different tokens of different tokens
+    pub fn get_withdraw_fee_percentage_per_silo_per_token(
+        &self,
+        silo_account: &AccountId,
+        token: &String,
+    ) -> u128 {
+        let silo_fee = self.withdraw_fee_percentage_per_silo.get(silo_account);
+        match silo_fee {
+            Some(token_fee) => token_fee
+                .token_to_fee_percentage
+                .get(token)
+                .unwrap_or(token_fee.default_fee_percentage),
+            None => {
+                return 0;
+            }
+        }
+    }
+
+    //token here should be ethereum address
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_deposit_fee_bound(
+        &mut self,
+        token: &String,
+        upper_bound: u128,
+        lower_bound: u128,
+    ) -> DepositTokenBounds {
+        let _ = validate_eth_address(token.clone());
+        self.deposit_fee_bound.insert(
+            token,
+            &DepositTokenBounds {
+                lower_bound,
+                upper_bound,
+            },
+        );
+        DepositTokenBounds {
+            lower_bound,
+            upper_bound,
+        }
+    }
+
+    //this should be added as per: 10% -> 0.1 = 0.1*10^6
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_deposit_fee_percentage(
+        &mut self,
+        token: &String,
+        eth_to_near: u128,
+        eth_to_aurora: u128,
+    ) -> DepositFeePercentage {
+        let _ = validate_eth_address(token.clone());
+        self.deposit_fee_percentage.insert(
+            token,
+            &DepositFeePercentage {
+                eth_to_near,
+                eth_to_aurora,
+            },
+        );
+        DepositFeePercentage {
+            eth_to_near,
+            eth_to_aurora,
+        }
+    }
+
+    // sets fee bounds for withdraw tokens, default bound: None
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_withdraw_fee_bound(
+        &mut self,
+        token: &String,
+        upper_bound: u128,
+        lower_bound: u128,
+    ) -> WithdrawTokenBounds {
+        let _ = validate_eth_address(token.clone());
+        self.withdraw_fee_bound.insert(
+            token,
+            &WithdrawTokenBounds {
+                lower_bound,
+                upper_bound,
+            },
+        );
+        WithdrawTokenBounds {
+            lower_bound,
+            upper_bound,
+        }
+    }
+
+    //this should be added as per: 10% -> 0.1 = 0.1*10^6
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_withdraw_fee_percentage(
+        &mut self,
+        token: &String,
+        near_to_eth: u128,
+        aurora_to_eth: u128,
+    ) -> WithdrawFeePercentage {
+        let _ = validate_eth_address(token.clone());
```

Comment:
I think it will be good to implement ser/des for EthAddress type and pass it instead of a string.
Here is the [implementation](https://github.com/aurora-is-near/fast-bridge-common/blob/master/src/lib.rs#L13) , you can move it to the `bridge-common` crate.